c++/src/kj/compat/tls.c++

```diff
@@ -407,29 +417,31 @@ private:
 // =======================================================================================
 // Implementations of ConnectionReceiver, NetworkAddress, and Network as wrappers adding TLS.
 
-class TlsConnectionReceiver final: public kj::ConnectionReceiver {
+class TlsConnectionReceiver final: public ConnectionReceiver, public TaskSet::ErrorHandler {
 public:
-  TlsConnectionReceiver(TlsContext& tls, kj::Own<kj::ConnectionReceiver> inner)
-      : tls(tls), inner(kj::mv(inner)) {}
+  TlsConnectionReceiver(TlsContext& tls, Own<ConnectionReceiver> inner)
+      : tls(tls), inner(kj::mv(inner)), tasks(*this) {
+    // Start up the connection loop.
+    tasks.add(acceptLoop());
+  }
+
+  void taskFailed(Exception&& e) override {
```

Comment:
FWIW I usually inherit privately from `TaskSet::ErrorHandler`, and declare `taskFailed()` as a private method, to avoid making it part of the public interface. But this whole class is private to this translation unit anyway so it doesn't really matter here. (No need to action this comment.)